#! /bin/sh
patch -p1 -f -l $* < $0
exit $?

From b510e73f4dabb5362e0e504cfe73318b975f235b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Tue, 28 Oct 2025 10:15:13 +0100
Subject: [PATCH] a11y: Raise UIA event for (un)checked item view items

This enables assistive tools on Windows to inform users when the
checked state of checkable item view items change.

Change-Id: I680209d3f5835c58bd91dd5826566c39ae72d7b6
---
 .../windows/uiautomation/qwindowsuiamainprovider.cpp          | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp b/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
index fc0e053f396..4fd9f8c6293 100644
--- a/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
+++ b/src/plugins/platforms/windows/uiautomation/qwindowsuiamainprovider.cpp
@@ -88,8 +88,8 @@ void QWindowsUiaMainProvider::notifyStateChange(QAccessibleStateChangeEvent *eve
 {
     if (QAccessibleInterface *accessible = event->accessibleInterface()) {
         if (event->changedStates().checked || event->changedStates().checkStateMixed) {
-           // Notifies states changes in checkboxes.
-           if (accessible->role() == QAccessible::CheckBox) {
+            // Notifies states changes in checkboxes and checkable item view items.
+            if (accessible->role() == QAccessible::CheckBox || accessible->role() == QAccessible::Cell || accessible->role() == QAccessible::ListItem || accessible->role() == QAccessible::TreeItem) {
                 if (QWindowsUiaMainProvider *provider = providerForAccessible(accessible)) {
                     VARIANT oldVal, newVal;
                     clearVariant(&oldVal);
--
2.51.0
